### PR TITLE
Update gouging settings

### PR DESCRIPTION
**Pull request by peterjan**
This PR removes the `migrationSurchargeMultiplier` from the gouging settings.

### CODE REVIEW

File: internal/gouging/gouging.go
Comment (ChrisSchinnerl) [inline internal/gouging/gouging.go:225]: We can hardcode the leeway to 12 blocks but I don't think we can remove the check. Nothing protects us from overpaying on uploads without it. 
For that reason, the block height is one of the relatively few fields of the price table that made it into rhp4 prices as well.

Comment (ChrisSchinnerl) [reply internal/gouging/gouging.go:225]: Another thing to keep in mind is that the v2 feature branch heavily changes gouging settings. So any change here will introduce massive conflicts. So we might be better off refactoring certain less urgent parts after the v2 code is merged.

Comment (peterjan) [reply internal/gouging/gouging.go:225]: I hardcoded a default of 12. The updated PR removes a function and swaps a setting for a default, so hopefully that won't massively conflict... I understand that would be annoying though, I'll leave it up to you on whether you think it'll be too much of a pain. Getting rid of this setting allows us to easily get rid of `WithGougingChecker` which makes it easy to fetch gouging parameters on the fly and only use the cache for usable hosts.

Comment (peterjan) [reply internal/gouging/gouging.go:225]: It would indeed conflict on [this update](https://github.com/SiaFoundation/renterd/pull/1418/files#diff-eb06058a1f2ee16c08a6db3d0cf6be2125e6cc385ccd2eb118d9df982a719c92R283), since its-happening is very up-to-date with `dev` the merge wouldn't be that much of a pain

File: stores/sql/mysql/migrations/main/migration_00029_update_gouging_settings.sql
Comment (peterjan) [reply stores/sql/mysql/migrations/main/migration_00029_update_gouging_settings.sql:1]: It's already tested and we're going to (hopefully) squash our migrations anyway. I don't mind removing it though 😁 I'll definitely keep it in mind next time but I feel that in general we've been cleaning up after ourselves, especially if it's just a `JSON_REMOVE`.

File: worker/pricetables.go
Comment (peterjan) [reply worker/pricetables.go]: Hm ok well at this point I'd rather keep it to removing the migration surcharge multiplier and cleaning up the `overpay` value in the download code. If we have to switch from setting to a default and we can't remove `BlocksUntilBlockHeightGouging` it essentially defeats the purpose of eventually restricting gouging solely to the upload code.

Comment (peterjan) [reply worker/pricetables.go]: Also fine with removing the setting, using the default and re-adding this check if you feel the setting should be removed. Just figured you'd prefer it this way.

**Review by ChrisSchinnerl — COMMENTED** [review]
This requires rebasing

File: .changeset/remove_migrationsurchargemultiplier_from_the_gouging_settings.md
Comment (alexfreska) [inline .changeset/remove_migrationsurchargemultiplier_from_the_gouging_settings.md:5]: `migrationSurchargeMultiplier`

**Review by ChrisSchinnerl — APPROVED** [review]

**Pull request merged**
